spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -61,77 +66,61 @@ class DLEstimator[@specialized(Float, Double) T: ClassTag](
 
   def setBatchSize(value: Int): this.type = set(batchSize, value)
 
+  /**
+   * number of max Epoch for the training
+   */
   val maxEpoch = new IntParam(this, "maxEpoch", "number of max Epoch", ParamValidators.gt(0))
-  setDefault(maxEpoch -> 20)
+  setDefault(maxEpoch -> 100)
 
   def getMaxEpoch: Int = $(maxEpoch)
 
   def setMaxEpoch(value: Int): this.type = set(maxEpoch, value)
 
   override def transformSchema(schema : StructType): StructType = {
-    validateAndTransformSchema(schema)
+    validateSchema(schema)
+    SchemaUtils.appendColumn(schema, $(predictionCol), ArrayType(DoubleType, false))
   }
 
-  protected override def internalFit(dataFrame: DataFrame): DLTransformerBase = {
-    val batches = toMiniBatch(dataFrame)
+  protected override def internalFit(
+      featureAndLabel: RDD[(Seq[AnyVal], Seq[AnyVal])]): DLTransformerBase = {
+    val batches = toMiniBatch(featureAndLabel)
     val dataset = DataSet.rdd(batches)
 
     val optimizer = Optimizer(model, dataset, criterion)
-      .setOptimMethod(new Adam[T]())
+      .setOptimMethod(new LBFGS[T]())
```

Comment:
Not sure if it's needed, but as maxEpoch, do you think we should provide interface for users to specify optim method as well ? Assuming for different tasks , different optim methods may be needed